src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -2642,8 +2643,326 @@ public override string TypeNameOfValue
+                case 61: return typeof(MethodGroup<,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,>).MakeGenericType(types[0], types[1], types[2], types[3], types[4], types[5], types[6], types[7], types[8], types[9], types[10], types[11], types[12], types[13], types[14], types[15], types[16], types[17], types[18], types[19], types[20], types[21], types[22], types[23], types[24], types[25], types[26], types[27], types[28], types[29], types[30], CreateMethodGroup(types, 31, 30));
+                case 62: return typeof(MethodGroup<,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,>).MakeGenericType(types[0], types[1], types[2], types[3], types[4], types[5], types[6], types[7], types[8], types[9], types[10], types[11], types[12], types[13], types[14], types[15], types[16], types[17], types[18], types[19], types[20], types[21], types[22], types[23], types[24], types[25], types[26], types[27], types[28], types[29], types[30], CreateMethodGroup(types, 31, 31));
+                case 63: return typeof(MethodGroup<,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,>).MakeGenericType(types[0], types[1], types[2], types[3], types[4], types[5], types[6], types[7], types[8], types[9], types[10], types[11], types[12], types[13], types[14], types[15], types[16], types[17], types[18], types[19], types[20], types[21], types[22], types[23], types[24], types[25], types[26], types[27], types[28], types[29], types[30], CreateMethodGroup(types, 31, 32));
+
+            }
+            throw Assert.Unreachable;
+        }
     }
 
+
+
+    class PSOutParameter<T> { private PSOutParameter() { }}
+
+    abstract class PSGenericType
+    {
+
+        public static Type GetGenericType(int i)
+        {
+            switch (i)
+            {
+                case 0: return typeof(PSGenericType0);
+                case 1: return typeof(PSGenericType1);
+                case 2: return typeof(PSGenericType2);
+                case 3: return typeof(PSGenericType3);
+                case 4: return typeof(PSGenericType4);
+                case 5: return typeof(PSGenericType5);
+                case 6: return typeof(PSGenericType6);
+                case 7: return typeof(PSGenericType7);
+                case 8: return typeof(PSGenericType8);
+                case 9: return typeof(PSGenericType9);
+                case 10: return typeof(PSGenericType10);
+                case 11: return typeof(PSGenericType11);
+                case 12: return typeof(PSGenericType12);
+                case 13: return typeof(PSGenericType13);
+                case 14: return typeof(PSGenericType14);
+                case 15: return typeof(PSGenericType15);
+                case 16: return typeof(PSGenericType16);
+            }
+            throw Assert.Unreachable;
```

Comment:
Real error here - and must resist urge to think of template metaprogramming to solve this problem.

Reply:
yeah - templates != generics. Or do you have some tricks up your sleave that I don't know about?

Hardcoded up to a limit, and dynamic type generation after that?

Reply:
Yeah, but hard coded is fine. MutableTuple has an interesting approach if you want to explore it - basically you have:
PSGenericType<T>
PSGenericType<T1, T2>
PSGenericType<T1, T2, T3, T4>
PSGenericType<T1, T2, T3, T4, T5, T6, T7, T8>
etc.

And if your list is too long, you encode the last parameter as another PSGenericType<...>.
If you don't have a power of 2, you use placeholder types like you've done with `Unit`.

Reply:
I actually already do that to an extent with MethodGroup. Maybe I could rewrite it to be more general.